x-pack/plugins/apm/server/lib/metrics/types.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ChartType, YUnit } from '../../../typings/timeseries';
+import { Coordinate } from '../../../typings/timeseries';
+
+export interface AggValue {
+  value: number | null;
+}
+
+export interface TimeSeriesBucket {
+  key_as_string: string; // timestamp as string
+  key: number; // timestamp as epoch milliseconds
+  doc_count: number;
+}
+
+export interface MetricsKeys {
+  [key: string]: AggValue;
+}
+
+interface SeriesDetails {
+  title: string;
+  color?: string;
+}
+
+type SeriesDetailsMap<T extends MetricsKeys> = {
+  [key in keyof T]: SeriesDetails
+};
+
+export interface ChartBase<T extends MetricsKeys> {
+  title: string;
+  key: string;
+  type: ChartType;
+  yUnit: YUnit;
+  transformValue?: (value: number) => number;
```

Comment:
Ah I added this because I was originally preparing the values on the server side (like transforming bytes to GB) but in the end that didn't make sense. I'll remove thanks.